### PR TITLE
EDSC-2923: Support creating subscriptions

### PR DESCRIPTION
**Pull request by trevorlang**
# Overview

### What is the feature?

Adds support for creating subscriptions

### What is the Solution?

Adds a panel when the user is on the "/search/granules/subscriptions" route that enables them to view current subscriptions, as well as create new ones.

### What areas of the application does this impact?

The navigation to the panel and the panel itself

# Testing

### Reproduction steps

- **Environment for testing:** SIT
- **Collection to test with:** Any collection/account with the proper ACLs

### Attachments

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: static/src/js/components/GranuleResults/GranuleResultsActions.js [hunk 1]
```diff
@@ -42,33 +45,30 @@ const GranuleResultsActions = ({
...
+  subscriptions
 }) => {
   const addToProjectButton = (
     <Button
       className="granule-results-actions__proj-action granule-results-actions__proj-action--add"
       onClick={() => onAddProjectCollection(focusedCollectionId)}
-      variant="link"
-      bootstrapVariant="link"
-      icon="plus-circle"
+      icon=""
```
Comment (trevorlang) [inline static/src/js/components/GranuleResults/GranuleResultsActions.js:55]: Remove

File: static/src/js/components/SearchPanels/SearchPanels.js [hunk 2]
```diff
@@ -325,6 +331,19 @@ class SearchPanels extends PureComponent {
       )
     }
 
+    const subscriptionsMoreActionsItem = isLoggedIn
```
Comment (trevorlang) [inline static/src/js/components/SearchPanels/SearchPanels.js:334]: Can we get rid of turnary

File: static/src/js/components/Subscriptions/SubscriptionsListItem.scss [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    margin-right: 0.125rem;
+    font-size: 0.75rem;
+  }
+
+  &__tooltip-query-list-item-value {
+    text-overflow: ellipsis;
+    overflow: hidden;
+    white-space: nowrap;
+  }
+}
```
Comment (trevorlang) [inline static/src/js/components/Subscriptions/SubscriptionsListItem.scss:76]: Add line

File: static/src/js/components/Subscriptions/SubscriptionsListItem.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                  }
+                </ul>
+              </Tooltip>
+            )}
+          >
+            <span className="subscriptions-list-item__query-text">
+              {
+                Object.keys(parsedQuery).map((key, i) => {
+                  const humanizedKey = humanizedGranuleQueryMap[key]
+
```
Comment (trevorlang) [inline static/src/js/components/Subscriptions/SubscriptionsListItem.js:62]: See if join would work here

File: static/src/js/components/SubscriptionsList/SubscriptionsList.js [hunk 5]
```diff
@@ -77,14 +83,35 @@ export const SubscriptionsList = ({
...
                           {name}
                         </td>
-                        <td>
+                        <td className="subscriptions-list-table__collection">
                           {title}
                         </td>
-                        <td>
-                          {query}
+                        <td className="subscriptions-list-table__query">
+                          <ul className="subscriptions-list-table__query-list">
```
Comment (trevorlang) [inline static/src/js/components/SubscriptionsList/SubscriptionsList.js:93]: See if we can clean up this page

File: static/src/js/components/SubscriptionsList/SubscriptionsList.scss [hunk 6]
```diff
@@ -31,4 +31,20 @@
...
+    white-space: nowrap;
+    overflow: hidden;
+  }
+
+  &__query-list {
+    margin: 0;
+    padding: 0;
+    list-style: none;
+  }
 }
```
Comment (trevorlang) [inline static/src/js/components/SubscriptionsList/SubscriptionsList.scss:50]: Add line

File: static/src/js/containers/AuthRequiredContainer/AuthRequiredContainer.js [hunk 7]
```diff
@@ -29,17 +30,22 @@ export class AuthRequiredContainer extends Component {
     const returnPath = window.location.href
 
     if (token === null || token === '') {
-      window.location.href = `${apiHost}/login?ee=${earthdataEnvironment}&state=${encodeURIComponent(returnPath)}`
+      this.setState({ isLoggedIn: false })
```
Comment (trevorlang) [inline static/src/js/containers/AuthRequiredContainer/AuthRequiredContainer.js:33]: if (redirect)...

File: static/src/js/containers/AuthRequiredContainer/AuthRequiredContainer.js [hunk 8]
```diff
@@ -53,7 +59,12 @@ export class AuthRequiredContainer extends Component {
   }
 }
 
+AuthRequiredContainer.defaultProps = {
+  redirect: true
+}
+
 AuthRequiredContainer.propTypes = {
+  redirect: PropTypes.bool,
```
Comment (trevorlang) [inline static/src/js/containers/AuthRequiredContainer/AuthRequiredContainer.js:67]: noRedirect?

File: static/src/js/selectors/query.js [hunk 9]
```diff
@@ -26,3 +30,21 @@ export const getFocusedCollectionGranuleQuery = createSelector(
     return collectionGranuleQuery
   }
 )
+
+export const getFocusedGranuleQueryString = createSelector(
+  [state => state, getFocusedCollectionMetadata],
+  (state, collectionMetadata) => {
+    const { collectionId } = collectionMetadata
+    // Extract granule search parameters from redux specific to the focused collection
+    const extractedGranuleParams = extractGranuleSearchParams(state, collectionId)
```
Comment (trevorlang) [inline static/src/js/selectors/query.js:39]: Can we improve this at all now?

File: static/src/js/util/__tests__/pruneFilters.test.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
+import { pruneFilters } from '../pruneFilters'
+
+describe('pruneFilters', () => {
+  test('removes ', () => {
```
Comment (trevorlang) [inline static/src/js/util/__tests__/pruneFilters.test.js:4]: Fix comment: Removes empty values

File: static/src/js/util/humanizedGranuleQueryMap.js [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+  circle: 'Circle',
+  browseOnly: 'Browse Only',
+  cloudCover: 'Cloud Cover',
+  collectionId: 'Collection ID',
+  dayNightFlag: 'Day/Night',
+  equatorCrossingDate: 'Equitorial Crossing Date',
+  equatorCrossingLongitude: 'Equitorial Longitude',
+  exclude: 'Exclude',
+  gridCoords: 'Grid Coordinates',
+  isCwic: 'Is CWIC',
```
Comment (trevorlang) [inline static/src/js/util/humanizedGranuleQueryMap.js:13]: Remove this

File: static/src/js/util/subscriptions.js [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+ */
+export const prepareSubscriptionQuery = (params) => {
+  const prunedParams = pruneFilters(params)
+
+  // Remove fields that should not be included in a subscription query
+  delete prunedParams.collectionId
+  delete prunedParams.conceptId
+  delete prunedParams.pageNum
+  delete prunedParams.sortKey
+
```
Comment (trevorlang) [inline static/src/js/util/subscriptions.js:16]: delete prunedParams.isCwic

File: static/src/js/components/Subscriptions/SubscriptionsBody.js [hunk 13]
```diff
@@ -0,0 +1,137 @@
+/* eslint-disable jsx-a11y/anchor-is-valid */
```
Comment (trevorlang) [inline static/src/js/components/Subscriptions/SubscriptionsBody.js:1]: Remove

**Review by macrouch — APPROVED** [review]

**Review by abbottry — APPROVED** [review]

**Pull request merged**